db/models/model.js

```diff
@@ -1,11 +1,34 @@
 const db = require("../connection");
+// const { articleData } = require("../data/test-data");
 
 const fetchTopicsFromDatabase = () => {
   return new Promise((resolve, reject) => {
-    db.query("SELECT * FROM topics;").then((queryResult) => {
-      resolve(queryResult.rows);
-    });
+    db.query("SELECT * FROM topics;")
+      .then((queryResult) => {
+        resolve(queryResult.rows);
+      })
+      .catch((error) => {
+        reject(error);
+      });
   });
 };
 
-module.exports = fetchTopicsFromDatabase;
+function fetchArticleFromDatabase(articleId) {
+  return new Promise((resolve, reject) => {
+    db.query("SELECT * FROM articles WHERE article_id=$1;", [articleId])
+      .then((queryResult) => {
+        if (queryResult.rows.length > 0) {
+          resolve(queryResult.rows);
+        } else {
+          console.log("No article found with articleId:", articleId);
+          resolve(null);
+        }
+      })
+      .catch((error) => {
```

Comment:
You could have one catch block in the controller instead to deal with rejected promises.